src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1170,7 +1170,7 @@ protected virtual void ReportUnassigned(Symbol symbol, SyntaxNode node, int slot
 
             if (skipIfUseBeforeDeclaration &&
                 symbol.Kind == SymbolKind.Local &&
-                (symbol.Locations.Length == 0 || node.Span.End < symbol.GetFirstLocationOrNone().SourceSpan.Start))
+                (symbol.TryGetFirstLocation() is var location && (location is null || node.Span.End < location.SourceSpan.Start)))
```

Comment:
📝 This is going to conflict with #67881 (where `TryGetFirstLocation` is eliminated)